shared/wallets/transaction/index.js

```diff
@@ -0,0 +1,180 @@
+  counterpartyType: CounterpartyType,
+|}
+
+const Icon = (props: IconProps) => {
+  switch (props.counterpartyType) {
+    case 'keybaseUser':
+      return <Avatar username={props.counterparty} size={props.large ? 48 : 32} />
+    case 'stellarPublicKey':
+      // TODO: Return anonymous user icon.
+      return null
+    case 'wallet':
+      // TODO: Return wallet icon.
+      return null
+    default:
+      /*::
+      declare var ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove: (counterpartyType: empty) => any
+      ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove(props.counterpartyType);
+      */
+      return null
+  }
+}
+
+type DetailProps = {|
+  large: boolean,
+  yourRole: Role,
+  counterparty: string,
+  counterpartyType: CounterpartyType,
+  amountUser: string,
+|}
+
+const Detail = (props: DetailProps) => {
+  const textType = props.large ? 'Body' : 'BodySmall'
+  const textTypeSemibold = props.large ? 'BodySemibold' : 'BodySmallSemibold'
+
+  let counterparty
+  switch (props.counterpartyType) {
+    case 'keybaseUser':
+      // TODO: Color counterparty based on following status.
+      counterparty = <Text type={textTypeSemibold}>{props.counterparty}</Text>
+      break
+    case 'stellarPublicKey':
+      counterparty = props.counterparty.substr(0, 6) + '...' + props.counterparty.substr(-5)
+      break
+    case 'wallet':
+      counterparty = props.large ? (
+        <Text type={textType}>{props.counterparty}</Text>
+      ) : (
+        <Text type={'BodySmallItalic'}>{props.counterparty}</Text>
+      )
+      break
+    default:
+      /*::
+      declare var ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove: (counterpartyType: empty) => any
+      ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove(props.counterpartyType);
+      */
+      break
+  }
+
+  const amount = <Text type={textTypeSemibold}>{props.amountUser}</Text>
+
+  if (props.counterpartyType === 'wallet') {
+    if (props.yourRole === 'sender') {
+      return (
+        <Text type={textType}>
+          You transferred Lumens worth {amount} from this wallet to {counterparty}.
+        </Text>
+      )
+    }
+
+    return (
+      <Text type={textType}>
+        You transferred Lumens worth {amount} from {counterparty} to this wallet.
+      </Text>
+    )
+  }
+
+  if (props.yourRole === 'sender') {
+    return (
+      <Text type={textType}>
+        You sent Lumens worth {amount} to {counterparty}.
+      </Text>
+    )
+  }
+
+  return (
+    <Text type={textType}>
+      {counterparty} sent you Lumens worth {amount}.
+    </Text>
+  )
+}
+
+type AmountXLMProps = {|
+  yourRole: Role,
+  amountXLM: string,
+  pending: boolean,
+|}
+
+const AmountXLM = (props: AmountXLMProps) => {
+  const color = props.pending
+    ? globalColors.black_20
+    : props.yourRole === 'sender' ? globalColors.red : globalColors.green
+  const amount = `${props.yourRole === 'sender' ? '-' : '+'} ${props.amountXLM}`
```

Comment:
Is there a better way to make the right side not wrap? `lineClamp={1}` makes it elide text with `...` instead.

Reply:
@cecileboucheron is working on mobile styling and is just putting the value on its own line so it doesn't interfere with the markdown. i think we should do that on desktop also (at least short term). sound ok cecile?
if we really want to keep it here in desktop you can use wrap: no-wrap i think. could likely do this with flow also making the left side have flex-shrink and this text have flex-shrink: 0

Reply:
@chrisnojima Sounds good to me to have the value on its own line. This is where it should be: 
![image](https://user-images.githubusercontent.com/2807426/39938194-05091bc2-5507-11e8-9baf-bfe344e9f2dc.png)